Source/CombatExtended/CombatExtended/Jobs/WorkGiver_ReloadTurret.cs

```diff
@@ -61,20 +65,21 @@ public override bool ShouldSkip(Pawn pawn, bool forced = false)
         public override bool HasJobOnThing(Pawn pawn, Thing t, bool forced = false)
         {
             // Should never happen anymore, as only TurretGunCEs should be returned from PotentialWorkThingsGlobal
-            if (!(t is Building_TurretGunCE)) {
+            Building_Turret turret = t as Building_Turret;
+            if (turret == null) {
```

Comment:
Is there a reason for this check ? The only items stored in the turret tracker mapcomp are turrets of the same type.

Reply:
I believe it is historical, given the comment on line 67.  I have updated the comment and condensed the cast+null-check.  We have to do the cast anyway, as the supercall's HasJobOnThing specifies `Thing` as the incoming type.  If the null check really is redundant, I'm hopeful that the JIT will ellide it (I believe it should).